server/api.go

```diff
@@ -167,6 +185,28 @@ func (a *apiServer) Publish(ctx context.Context, req *client.PublishRequest) (
 	}
 	a.logger.Debugf("api: Publish [subject=%s]", subject)
 
+	stream := a.metadata.GetStream(req.Stream)
+	if stream == nil {
```

Comment:
It just occurred to me that this won't work for [`PublishToSubject`](https://github.com/liftbridge-io/go-liftbridge/blob/master/client.go#L181-L192) which publishes Liftbridge messages directly to a NATS subject. That's probably OK since the expectation is that `Publish` operates at a stream level while `PublishToSubject` doesn't.

In the case of `PublishToSubject`, `Stream` will be `nil` on the `PublishRequest` since `Subject` is set instead. This now has me wondering if `Publish` and `PublishToSubject` should just be separate explicit endpoints.

Reply:
Indeed. I've added a check for an empty stream name in 4a99036.

I do personally find it a bit counter intuitive to have two functions that are very similarly named but have a very different set of constraints/limitations.